libnetdata/config/appconfig.c

```diff
@@ -42,6 +44,78 @@ struct section {
                             // readers are protected using the rwlock in avl_tree_lock
 };
 
+struct _connector_instance {
+    struct section  *connector;        // actual connector
+    struct section  *instance;         // This instance
+    struct _connector_instance *next;   // Next instance
+};
+
+struct config exporting_config = {
+        .sections = NULL,
+        .mutex = NETDATA_MUTEX_INITIALIZER,
+        .index = {
+                .avl_tree = {
+                        .root = NULL,
+                        .compar = appconfig_section_compare
+                },
+                .rwlock = AVL_LOCK_INITIALIZER
+        }
+};
+
+struct _connector_instance  *ci = NULL;
+
+int get_connector_instance(struct connector_instance *target_ci) {
+    static struct _connector_instance *local_ci = NULL;
+
+    if (unlikely(!ci))
+        return 0;
+
+    if (target_ci == NULL) {
+        local_ci = ci;
+        return 1;
+    }
+    if (local_ci == NULL)
+        local_ci = ci;
+    else {
+        local_ci = local_ci->next;
+        if (local_ci == NULL)
+            return 0;
+    }
+
+    strcpy(target_ci->instance_name, local_ci->instance->name);
+    strcpy(target_ci->connector_name, local_ci->connector->name);
+
+    return 1;
+}
+
+int is_valid_connector(char *type){
+    if (unlikely(!type))
+        return 0;
+
+    if(!strcmp(type, "graphite") || !strcmp(type, "graphite:plaintext")) {
+            return 1;
+        }
+        else if(!strcmp(type, "opentsdb") || !strcmp(type, "opentsdb:telnet")) {
```

Comment:
All `else  if` statement are not correct aligned.

Reply:
I will fix it -- compiler didn't complain